iModelJsNodeAddon/presentation/ECPresentationSerializer.cpp

```diff
@@ -907,7 +911,9 @@ NavNodeKeyPtr IModelJsECPresentationSerializer::_GetNavNodeKeyFromJson(IConnecti
 NavNodeKeyPtr IModelJsECPresentationSerializer::_GetBaseNavNodeKeyFromJson(BeJsConst json) const
     {
     Utf8CP type = json["type"].asCString();
-    return NavNodeKey::Create(type, "", ParseNodeKeyHashPath(json["pathFromRoot"]));
+    NavNodeKeyPtr key = NavNodeKey::Create(type, "", ParseNodeKeyHashPath(json["pathFromRoot"]));
+    key->SetInstanceKeysSelectQuery(std::move(GetPresentationQueryFromJson(json["instanceKeysSelectQuery"])));

```

Comment:
don't need to use `std::move` for rvalues

Reply:
also, what happens if `json` doesn't have the attribute?

Reply:
Removed the `std::move`.
It creates an empty query. Added a check to avoid it.

Reply:
If by "empty" you mean null, then there's no point in adding a null check and you can make a straight assignment. If "empty" means a query object with empty string, then your null check will pass and you're still going to assign it to the node. So needs changes in any case.

Reply:
Maybe `GetPresentationQueryFromJson` should just return `nullptr` if:
- the given JSON is not an object (e.g. null) 
- doesn't have the required `"query"` member
- the `"query"` member is not a string or is an empty string